src/nvim/option.c

```diff
@@ -2649,24 +2649,40 @@ static char *did_set_string_option(int opt_idx, char_u **varp, bool new_value_al
       }
       s = skip_to_option_part(s);
     }
-  } else if (varp == &p_lcs) {  // 'listchars'
+  } else if (varp == &p_lcs) {  // global 'listchars'
     errmsg = set_chars_option(curwin, varp, false);
-    if (!errmsg) {
+    if (errmsg == NULL) {
+      // The current window is set to use the global 'listchars' value.
+      // So clear the window-local value.
+      if (!(opt_flags & OPT_GLOBAL)) {
+        clear_string_option(&curwin->w_p_lcs);
+      }
       FOR_ALL_TAB_WINDOWS(tp, wp) {
-        set_chars_option(wp, &wp->w_p_lcs, true);
+        errmsg = set_chars_option(wp, &wp->w_p_lcs, true);
+        if (errmsg) {
```

Comment:
Theoretically this `if` statement is useless (it uses `break;` which only breaks out of inner loop) and redundant (invalid values have been checked before). I'm considering opening a PR to Vim.